File: scripts/Find the Duplicate Number/Find the Duplicate Number.js
var findDuplicate = function(nums) {
    let dup = new Map();

    for(let val of nums){
        if(dup.has(val)){
            dup.set(val, dup.get(val) + 1);
            return val;
        }
        else
            dup.set(val, 1);
    }
};

File: algorithms/F/Find the Duplicate Number/Find the Duplicate Number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int i = 0; 
         while (i < nums.length) {

            if (nums[i] != i + 1) {
                int correct = nums[i] - 1;
                if (nums[i] != nums[correct]) {
                    swap(nums, i , correct);
                } else {
                    return nums[i];
                }
            } else {
                i++;
            }
        }
        return -1;
    }
     static void swap(int[] arr, int first, int second) {
        int temp = arr[first];
        arr[first] = arr[second];
        arr[second] = temp;
    }
   }
